rust_src/src/charset.rs

```diff
@@ -18,6 +23,21 @@ impl LispObject {
     }
 }
 
+/// Internal use only.
+/// Clear temporary charset mapping tables.
+/// It should be called only from temacs invoked for dumping.
+#[lisp_fn]
+pub fn clear_charset_maps() {
+    unsafe {
+        xfree(temp_charset_work as *mut libc::c_void);
```

Comment:
The pointer has to be checked first. Calling free on a NULL is not guaranteed to be safe. This is why the C code has an `if`.